models/measurements.py

```diff
@@ -359,15 +366,63 @@ def find_cutouts_in_list(self, cutouts_list):
 
         raise ValueError('Cutouts not found in the list. ')
 
-    def passes(self):
+    def compare_to_thresholds(self):
         """check if there are disqualifiers above the threshold
 
-        Note that if a threshold is missing or None, that disqualifier is not checked
+        Compares disqualifiers to two threshold dictionaries: 
+            - deletion_thresh: represents values above which the object
+              will be 'deleted' (not saved to database)
+            - mark_thresh: represents values above which the object 
+              will be have is_bad marked True (but saved)
+
+        If the deletion threshold in parameters is set to None, it will
+        use the regular threshold as the deletion threshold. If instead
+        the deletion threshold is an empty dictionary, no objects will
+        be marked "delete"
+        NOTE: Maybe this functionality is overkill? Instead, I could just
+        make it such that you cannot have deletion_thresholds None, but rather
+        it has to be a dictionary with all the keys set to a value of None to
+        trigger using the threshold as deletion_threshold.
+        This would work fine, but to never delete items you would have to
+        manually set impossible values rather than a simple empty dict.
+
+        This function sets the is_bad attribute/column to its proper 
+        values based on the above comparison.
+
+        returns one of three strings to indicate the result
+          - "ok"     : All disqualifiers below both thresholds
+          - "bad"    : Some disqualifiers above mark_thresh but all 
+                       below deletion_thresh
+          - "delete" : Some disqualifiers above deletion_thresh
+        """
+        passing_status = "ok"
+
+        mark_thresh = self.provenance.parameters['thresholds'] # thresholds above which measurement is marked 'bad'
+        deletion_thresh = (mark_thresh if self.provenance.parameters['deletion_thresholds'] is None
+                           else self.provenance.parameters['deletion_thresholds'])
+
+        for key, value in mark_thresh.items():
+            if deletion_thresh.get(key) is not None and self.disqualifier_scores[key] >= deletion_thresh[key]:
+                passing_status =  "delete"
+                break
+            if mark_thresh[key] is not None and self.disqualifier_scores[key] >= mark_thresh[key]:
+                # I didn't use get here since its required - should fire an error if key isn't in the dict I think
+                passing_status = "bad" # no break because another key could trigger "delete"
```

Comment:
In the init of `ParsMeasurer`, a value is set for the `thresholds` parameter that is hard coded in and then overwritten by the kwargs from the config file, so I guess this would be possible only if the user overwrote that dictionary with one missing keys. I can loop over both just in case.